File: app/channels/comments_channel.rb
class CommentsChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def follow(data)
    stream_from "#{data['commentable_type'].underscore}_#{data['commentable_id']}_comments"
  end
end

File: app/javascript/channels/comments_channel.js
import consumer from "channels/consumer"

let subscription_comment

function subscribeToComments() {
  const commentsList = document.querySelector(".comments")
  if (!commentsList) return

  if (subscription_comment) {
    consumer.subscriptions.remove(subscription_comment)
  }

  subscription_comment = consumer.subscriptions.create("CommentsChannel", {
    connected() {
      const comments = document.querySelector(".comments")
      if (!comments) return
      this.perform("follow", {
        commentable_type: comments.dataset.commentableType,
        commentable_id: comments.dataset.commentableId
      })
    },

    disconnected() {
      // Called when the subscription has been terminated by the server
    },

    received(data) {
      const comments = document.querySelector(".comments")
      if (!comments) return

      comments.insertAdjacentHTML(
          "beforeend",
          `<p>${data.body}</p>`
      )
    }
  })
}

document.addEventListener("turbo:load", subscribeToComments);
